blockchyp.go

```diff
@@ -198,6 +198,43 @@ func (client *Client) TC(request TermsAndConditionsRequest) (*TermsAndConditions
 
 }
 
+/*
+Balance retrives the balance of a payment method, if applicable.
+*/
+func (client *Client) Balance(request BalanceRequest) (*BalanceResponse, error) {
+
+	var response BalanceResponse
+
+	route, err := client.resolveTerminalRoute(request.TerminalName)
+	if err != nil {
+		if err == ErrUnknownTerminal {
+			response.Error = ResponseUnknownTerminal
+			return &response, err
+		}
+
+		return nil, err
+	}
+
+	if route.CloudRelayEnabled {
+		err = client.RelayRequest("/balance", http.MethodPost, request, &response, request.Test)
+	} else {
+		terminalRequest := TerminalBalanceRequest{
+			APICredentials: route.TransientCredentials,
+			Request:        request,
+		}
+		err = client.terminalPost(route, "/balance", terminalRequest, &response)
+	}
+
+	if err, ok := err.(net.Error); ok && err.Timeout() {
+		response.Error = ResponseTimedOut
+	} else if err != nil {
```

Comment:
err on line 230 is shadowed by err on line 228, which will potentially swallow errors and return nil, nil. You should change the name of the declaration on line 228. For example:
```
    if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
        response.Error = ResponseTimedOut
    } else if err != nil {
```

Reply:
fixed in the latest commit